kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ChecksumSimpleComparisonSuite.scala

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults
+
+import io.delta.kernel.Operation.CREATE_TABLE
+import io.delta.kernel.{Operation, Table, Transaction}
+import io.delta.kernel.data.{ColumnarBatch, Row}
+import io.delta.kernel.defaults.utils.TestUtils
+import io.delta.kernel.engine.Engine
+import io.delta.kernel.expressions.Literal
+import io.delta.kernel.internal.actions.{AddFile, SingleAction}
+import io.delta.kernel.internal.checksum.{CRCInfo, ChecksumReader}
+import io.delta.kernel.internal.fs.Path
+import io.delta.kernel.internal.util.{FileNames, VectorUtils}
+import io.delta.kernel.internal.util.Utils.{singletonCloseableIterator, toCloseableIterator}
+import io.delta.kernel.types.IntegerType.INTEGER
+import io.delta.kernel.types.StructType
+import io.delta.kernel.utils.CloseableIterable.{emptyIterable, inMemoryIterable}
+import io.delta.kernel.utils.{CloseableIterator, DataFileStatus, FileStatus}
+import org.scalatest.funsuite.AnyFunSuite
+
+import java.io.File
+import java.nio.file.Files
+import java.util
+import java.util.Collections.{emptyMap, singletonMap}
+import java.util.Optional
+import scala.collection.JavaConverters._
+
+/**
+ * Test suite to verify checksum file correctness by comparing
+ * Delta Spark and Delta Kernel generated checksum files.
+ * This suite ensures that both implementations generate consistent checksums
+ * for various table operations.
+ */
+class ChecksumSimpleComparisonSuite extends AnyFunSuite with TestUtils {
+
+  private val PARTITION_COLUMN = "part"
+
+  test("create table, insert data and verify checksum") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val sparkTablePath = tablePath + "spark"
+      val kernelTablePath = tablePath + "kernel"
+      Table
+        .forPath(engine, kernelTablePath)
+        .createTransactionBuilder(engine, "test-engine", CREATE_TABLE)
+        .withSchema(engine, new StructType().add("id", INTEGER))
+        .build(engine)
+        .commit(engine, emptyIterable())
+        .getPostCommitHooks
```

Comment:
Extend the DeltaWriteSuiteBase. It has utility methods to create the txn and other helper methods